account-kit/signer/src/session/manager.ts

```diff
@@ -170,10 +170,14 @@ export class SessionManager {
      * We should revisit this later
      */
     if (session.expirationDateMs < Date.now()) {
-      this.store.setState({ session: null });
+      this.clearSession();
       return null;
     }
 
+    setTimeout(() => {
+      this.clearSession();
+    }, session.expirationDateMs - Date.now());
```

Comment:
Slightly odd in that if `getSession` is called multiple times, then it will set this timeout multiple times, although I don't know if that actually causes any problems. Still, I would consider storing the timeout id and clearing the old timeout when setting a new one.

Reply:
Also: I don't know if this would be an issue, but there's a funny thing where if you set a timeout longer than 2^31, or around 25 days, then it triggers immediately. It may be worth putting in a check for that, in case someone asks for like 30 day session expirations.

Reply:
oh wow... TIL about setTimeout max being 24.8 days. yea I'll make sure to check that actually

re: getSession being called multiple times, yea good call. I'll add a global handler for this